compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1453,7 +1485,31 @@ public BType readType(int cpI) throws IOException {
                     }
                     int funcCount = inputStream.readInt();
                     for (int i = 0; i < funcCount; i++) {
-                        ignoreAttachedFunc();
+                        //populate intersection type object functions
+                        if (isImmutable(objectSymbol.flags) && Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
+                            String attachedFuncName = getStringCPEntryValue(inputStream);
+                            var attachedFuncFlags = inputStream.readLong();
+                            if (Symbols.isFlagOn(attachedFuncFlags, Flags.INTERFACE) &&
+                                    Symbols.isFlagOn(attachedFuncFlags, Flags.ATTACHED)) {
+                                BInvokableType attachedFuncType = (BInvokableType) readTypeFromCp();
+                                Name funcName = names.fromString(Symbols.getAttachedFuncSymbolName(
+                                        objectSymbol.name.value, attachedFuncName));
+                                BInvokableSymbol attachedFuncSymbol =
```

Comment:
When an implementation is not found for a function, the function name in the error seems to have additional info when it is via the BIR.

```ballerina
import ballerina/lang.runtime;

public type StackFrame readonly & object {
   public function toString() returns string;
};

readonly class Foo {
    *StackFrame; // no implementation found for the method 'toString' of class 'Foo'
}

readonly class Bar {
    *runtime:StackFrame; // no implementation found for the method '(ballerina/lang.runtime:0.0.1:$anonType$_0 & readonly).toString' of class 'Bar'
}
```

Reply:
Addressed in 1b6b21ff9a8ad716b114fd76ad04f19dc70a57e8